hub/core/version_control/commit_node.py

```diff
@@ -0,0 +1,34 @@
+from datetime import datetime
+from typing import List, Optional
+
+
+class CommitNode:
+    """Contains all the Version Control information about a particular commit."""
+
+    def __init__(self, branch: str, commit_id: Optional[str] = None):
+        self.commit_id = commit_id
+        self.branch = branch
+        self.children: List["CommitNode"] = []
+        self.parent: Optional["CommitNode"] = None
+        self.commit_message: Optional[str] = None
+        self.commit_time: Optional[datetime] = None
+        self.commit_user_name: Optional[str] = None
+
+    def add_child(self, node: "CommitNode"):
```

Comment:
This method is never used. Maybe remove this and rename add_successor to add_child?